go-update/src/main/java/org/reactome/release/goupdate/DuplicateReporter.java

```diff
@@ -81,23 +84,29 @@ public DuplicateReporter(MySQLAdaptor adaptor)
 			{
 				for (GKInstance i : instances)
 				{
-					for (GKSchemaAttribute referrerAttrib : (Set<GKSchemaAttribute>)i.getSchemClass().getReferers())
-					{
-						@SuppressWarnings("unchecked")
-						Collection<GKInstance> referrers = (Collection<GKInstance>) i.getReferers(referrerAttrib);
-						if (referrerCounts.containsKey(i.getDBID()))
-						{
-							referrerCounts.put(i.getDBID(), referrerCounts.get(i.getDBID()) + referrers.size());
-						}
-						else
-						{
-							referrerCounts.put(i.getDBID(), referrers.size());
-						}
-					}
+					Long dbid = i.getDBID();
+					int refCount = getReferrerCountforInstance(i, classesToIgnore);
+					referrerCounts.put(dbid,refCount);
 				}
 			}
 		}
 		return referrerCounts;
 	}
 	
+	public int getReferrerCountforInstance(GKInstance instance, String ... classesToIgnore) throws Exception
+	{
+		int refCount = 0;
+		
+		for (GKSchemaAttribute referrerAttrib : (Set<GKSchemaAttribute>) instance.getSchemClass().getReferers())
+		{
+			@SuppressWarnings("unchecked")
+			Collection<GKInstance> referrers = (Collection<GKInstance>) instance.getReferers(referrerAttrib);
+			if (classesToIgnore != null && classesToIgnore.length > 0)
+			{
+				referrers = referrers.stream().filter(r -> !(new HashSet<String>( Arrays.asList(classesToIgnore))).contains(r.getSchemClass().getName())).collect(Collectors.toList());
```

Comment:
It took me some time to understand this was checking the referrer class was not in the list of classes to ignore.

`!(new HashSet<String>( Arrays.asList(classesToIgnore))).contains(r.getSchemClass().getName())`

Could it be refactored to be more declarative?  Also, I think you can call contains directly on the List instead of wrapping in a HashSet.

Reply:
I'll have a look at this